tests/c/pthread_create.c

```diff
@@ -1,27 +1,86 @@
 // Compiler:
 // Run-time:
-//   status: error
-//   stderr: ...
-//    not yet implemented: Allocate and set thread-local ShadowStack instance. No support for threads yet!
+//   stderr:
 //    ...
+//    ret1 = 0xdead
+//    ...
+//    oof2: {{_}} 20
+//    ...
+//    ret2 = 0xbaff1ed
+
+// This test checks that new threads create their own shadow stack. For this is
+// creates two threads `foo` and `bar` in such a way that calling `overwrite`
+// on the main thread would assign it the same shadow stack as `bar`, if
+// threads do not have their own shadow stacks. Thus creating variables on the
+// stack of `overwrite` clobbers values on the stack of `bar`.
 
 #include <pthread.h>
 #include <yk.h>
 #include <yk_testing.h>
+#include <unistd.h>
+#include <stdatomic.h>
+
+atomic_int order;
+
+void overwrite() {
+  int a = 99;
+  int b = 99;
+  int c = 99;
+  int d = 99;
+  int e = 99;
+  int f = 99;
+  int g = 99;
+  int h = 99;
```

Comment:
Is this test x64-specific? I assume a..h are to fill the right number of registers so that spills happen?